apps/sms/test/marionette/lib/storage.js

```diff
@@ -57,15 +27,29 @@
          * pre-populate storage with.
          */
         setContactsStorage: function(contacts) {
-          client.executeScript(function(contacts) {
-            var STORAGE_NAME = 'contactsDB';
-
-            window.wrappedJSObject.TestStorages.setStorage(STORAGE_NAME, {
-              contacts: contacts
-            });
+          this.setStorage('contactsDB', {
+            contacts: contacts || []
+          });
+        },
 
-            window.wrappedJSObject.TestStorages.setStorageReady(STORAGE_NAME);
-          }, [contacts || []]);
+        /**
+         * Sets pre-populated custom storage.
+         * @param {string} storageName Name of the custom storage.
+         * @param {*} storage Custom storage data.
+         * @private
+         */
+        setStorage: function(storageName, storage) {
+          client.scope({ context: 'chrome' }).executeScript(
+            function(storage, storageName) {
+              Services.obs.addObserver(function(window) {
+                window.wrappedJSObject.TestStorages.set(
+                  storageName,
+                  Components.utils.cloneInto(storage, window.wrappedJSObject)
+                );
+              }, 'test-storage-ready', false);
```

Comment:
> so that means we can't set the storage after the event has been sent though ? Not a big deal though.

Nope, not currently (and we can not update storage in master as well), can be changed when we need such a feature.

> also can you unregister the observer ?

Not sure, but in this patch I don't really want to do this as a lot of tests require the same storage after app is relaunched. I think it can be part of the "reset/change app storage" if we need this eventually.